packages/circuit-ui/components/ImageInput/ImageInput.tsx

```diff
@@ -293,21 +326,71 @@ export const ImageInput = ({
     clearInputElement();
   };
 
+  const handlePaste = (event: ClipboardEvent) => {
+    const { files } = event.clipboardData;
+    handleChange(files);
+
+    if (inputRef.current && files) {
+      // An error is thrown when trying to assign anything but a FileList here.
+      // For security reasons, it's not possible to simulate a FileList object.
+      // That's why this code has to be disabled during testing.
+      if (process.env.NODE_ENV !== 'test') {
+        inputRef.current.files = files;
+      }
+    }
+  };
+
+  const handleDragging = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(true);
+  };
+
+  const handleDragLeave = (event: DragEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setDragging(false);
+  };
+
+  const handleDrop = (event: DragEvent) => {
+    handleDragLeave(event);
+    const files = event.dataTransfer?.files;
+    handleChange(files);
+
+    if (inputRef.current && files) {
+      // An error is thrown when trying to assign anything but a FileList here.
+      // For security reasons, it's not possible to simulate a FileList object.
+      // That's why this code has to be disabled during testing.
+      if (process.env.NODE_ENV !== 'test') {
+        inputRef.current.files = files;
+      }
+    }
+  };
+
   return (
     <Fragment>
-      <InputWrapper>
+      <InputWrapper onPaste={handlePaste}>
```

Comment:
Why does the handler need to go on the wrapper? Shouldn't it be on the focused element?(the input)

Reply:
That's what I thought too, but it didn't work. Only this did. In any case, the event bubbles up from the input.